Update _toc.yml.in

Rmalavally opened the pull request:
DRAFT - TOC restructure

Rmalavally committed:
Update _toc.yml.in

TOC restructure

samjwu commented on docs/sphinx/_toc.yml.in:
```diff
@@ -1,24 +1,44 @@
+      - file: MIOpen_Porting_Guide.rst
+      - file: citation.rst
+    
+    title: Quick-start
+    - file: reference/install.rst
```
```suggestion
- caption: Quick-start
  entries:
  - file: reference/install.rst
```

Rmalavally committed:
Update _toc.yml.in

Changed title to caption

Rmalavally committed:
Update _toc.yml.in

samjwu reviewed (dismissed):
The table of contents has a very specific format it accepts

A good example is the ROCm TOC: https://github.com/RadeonOpenCompute/ROCm/blob/develop/docs/sphinx/_toc.yml.in

samjwu commented on docs/sphinx/_toc.yml.in:
```diff
@@ -1,24 +1,44 @@
-  maxdepth: 6
+  maxdepth: 7
 root: index
+caption: What is MIOpen?  
```
```suggestion
- caption: What is MIOpen?
  entries:
  - file:
```

samjwu commented on docs/sphinx/_toc.yml.in:
```diff
@@ -1,24 +1,44 @@
+      - file: MIOpen_Porting_Guide.rst
+      - file: citation.rst
+    
+    caption: Quick-start
```
```suggestion
- caption: Quick-start
  entries:
  - file:
```

Rmalavally committed:
Update _toc.yml.in

Corrected white space for - caption

samjwu committed:
Update _toc.yml.in

samjwu committed:
Update _toc.yml.in

Rmalavally reviewed (commented):
Thanks, Sam.

bghimireamd commented:
LGTM

Rmalavally commented:
> LGTM

Thanks so much.

LisaDelaney commented on docs/sphinx/_toc.yml.in:
```diff
@@ -1,24 +1,34 @@
-  - file: MIOpen_Porting_Guide
-  - file: apireference
-- caption: About
+- caption: What is MIOpen?
```
this should not be a heading. it should be an intro page that describes the project.

Rmalavally replied:
We are still discussing the content for the Intro page. For now, we have reorganized the ToC on what's available and accurate. We have also removed the reference to release notes, as it is obsolete.

LisaDelaney replied:
then we need to wait until we have a page to link to. It's confusing to have a heading represent a placeholder for a future page. A heading (='caption' in the toc) should only be there to represent pages that fit into that heading.

Rmalavally replied:
That's a decision we must make. Working with the team.

LisaDelaney replied:
you can use the README file to get a little general info for a 'What is ..?' page until the team gets a chance to add to it. this page doesn't need to super long or elaborate, just has to describe what the project is/does.

Rmalavally replied:
Thanks, Lisa. Thanks so much for your help and comments.

LisaDelaney commented on docs/sphinx/_toc.yml.in:
where is the API ref heading? you seem to have removed the entire former API ref section and there is no organization to the existing pages

Rmalavally replied:
https://github.com/ROCmSoftwarePlatform/MIOpen/blob/Rmalavally-patch-1/docs/sphinx/_toc.yml.in

LisaDelaney replied:
look at the build--it's not rendering, so you need to fix the code.

Rmalavally replied:
Yep. It's a work in progress. Won't render yet.

LisaDelaney replied:
you haven't added any of the Doxygen content (=the API ref)

Rmalavally replied:
- file: reference/apireference.rst is what I have from them. Let's hold off until the structure is finalized.

LisaDelaney replied:
Doxygen builds content automatically from the header files (which contains the code/text/content for the API). The team does does not supply the API textual content. This content is build into the "Doxygen" folder--this is what you need to include in the toc in order to incorporate the API ref.

LisaDelaney commented:
these are no longer in the toc--is this intentional? Building MIOpen for Embedded Systems, Building the driver, Prerequisites

Rmalavally commented:
> these are no longer in the toc--is this intentional? Building MIOpen for Embedded Systems, Building the driver, Prerequisites

The restructure is based on their approval. I am not an expert to make this call. Let me finalize the content and get back to the team.

LisaDelaney commented:
> > these are no longer in the toc--is this intentional? Building MIOpen for Embedded Systems, Building the driver, Prerequisites
> 
> The restructure is based on their approval. I am not an expert to make this call. Let me finalize the content and get back to the team.

We can't publish their documentation without the API ref. The API ref is the most important part.

Rmalavally commented:
> > > these are no longer in the toc--is this intentional? Building MIOpen for Embedded Systems, Building the driver, Prerequisites
> > 
> > 
> > The restructure is based on their approval. I am not an expert to make this call. Let me finalize the content and get back to the team.
> 
> We can't publish their documentation without the API ref. The API ref is the most important part.

Sure. I am aware. We used to have MIOpen on RTD before with other docs. Let's not rush without ensuring we are doing it correctly.

LisaDelaney commented:
the landing page (index) is not ready. This PR needs more than toc edits before it can be merged...(at a minimum, add the updated index and what is pages)

Rmalavally commented:
> the landing page (index) is not ready. This PR needs more than toc edits before it can be merged...(at a minimum, add the updated index and what is pages)

I agree. Even with the minimum updates, it may not be consistent. Let's evaluate the implications of making this publicly available for this release.

Rmalavally committed:
Update _toc.yml.in

Removed citation.rst based on customer request

https://ontrack-internal.amd.com/browse/SWDEV-416591

samjwu commented on docs/sphinx/_toc.yml.in:
```diff
@@ -1,24 +1,34 @@
+  - file: reference/driver.rst
+- caption: API reference
+  entries:
+  - file: reference/apireference.rst
```
the files here in the TOC are relative to the `docs/` folder

for this to render correctly, `apireference.rst` needs to go into a folder named `reference`

samjwu replied:
same goes for tutorials

Rmalavally committed:
Update _toc.yml.in

Updates

Rmalavally committed:
Create reference

Folder for MIOpen reference

Rmalavally committed:
Delete docs/reference

Rmalavally committed:
Create quick-start.rst

Rmalavally committed:
Create what-is-MIOpen.rst

Adding What is MIOpen.rst

Rmalavally committed:
Delete docs/tutorials/quick-start.rst

Rmalavally committed:
Update what-is-MIOpen.rst

Rmalavally committed:
Update what-is-MIOpen.rst

Rmalavally committed:
Create apireference.rst

Adding MIOpen API documentation

Rmalavally committed:
Update apireference.rst

Add API reference documentation

Rmalavally committed:
Create find_and_immediate.md

Rmalavally committed:
Create finddb.rst

Rmalavally committed:
Update finddb.rst

Rmalavally committed:
Delete docs/tutorials/finddb.rst

Rmalavally committed:
Create finddb.md

Rmalavally committed:
Create cache.md

Rmalavally committed:
Create perfdatabase.md

Rmalavally committed:
Create getting_started_fusionAPI.md

Rmalavally committed:
Create debugging_and_logging.md

Rmalavally committed:
Create MI200alternateimplementation.md

Rmalavally committed:
Create MIOpenportingguide.md

Rmalavally committed:
Create index.rst

Rmalavally committed:
Create install.md

Rmalavally committed:
Create embed.md

Rmalavally committed:
Create driver.md

Rmalavally committed:
Update _toc.yml.in

Rmalavally committed:
Delete docs/DebugAndLogging.md

Rmalavally committed:
Delete docs/Getting_Started_FusionAPI.md

Rmalavally committed:
Delete docs/MI200AlternateImplementation.md

Rmalavally committed:
Delete docs/MIOpen_Porting_Guide.md

Rmalavally committed:
Delete docs/apireference.rst

Rmalavally committed:
Delete docs/cache.md

Rmalavally committed:
Delete docs/driver.md

Rmalavally committed:
Create index.rst

Rmalavally committed:
Delete docs/find_and_immediate.md

Rmalavally committed:
Delete docs/embed.md

Rmalavally committed:
Delete docs/finddb.md

Rmalavally committed:
Delete docs/install.md

Rmalavally committed:
Delete docs/perfdatabase.md

Rmalavally committed:
Update _toc.yml.in

Rmalavally committed:
Update _toc.yml.in

Rmalavally committed:
Update _toc.yml.in

Rmalavally committed:
Update _toc.yml.in

Add titles

Rmalavally committed:
Update _toc.yml.in

Rmalavally committed:
Update index.rst

Rmalavally committed:
Merge branch 'develop' into Rmalavally-patch-1

SwRaw committed:
Update .readthedocs.yaml

SwRaw committed:
Update _toc.yml.in

Rmalavally committed:
Update _toc.yml.in

Rmalavally committed:
Update _toc.yml.in

Rmalavally commented:
@samjwu Using 'caption' breaks the build. Without the use of 'caption', it builds fine. Please advise.

Rmalavally committed:
Update what-is-MIOpen.rst

JehandadKhan commented on docs/tutorials/what-is-MIOpen.rst:
```diff
@@ -0,0 +1,17 @@
+.. meta::
+   :description: MIOpen is AMD’s deep learning primitives library, which provides highly optimized, and hand-tuned implementations of
+   different operators such as convolution, batch normalization, pooling, softmax, activation and layers for Recurrent Neural
+      Networks (RNNs), used in both training and inference [9].
```
Please remove the reference.

Rmalavally replied:
I did. Let me resend the PR. Here's the final text,

MIOpen is AMD’s deep learning primitives library, which provides highly optimized and hand-tuned implementations of different operators such as convolution, batch normalization, pooling, softmax, activation and layers for Recurrent Neural Networks (RNNs), used in both training and inference. Moreover, MIOpen is fully open source, including all its GPU kernels, complementing AMD’s open source ROCm stack. MIOpen is the first to extend the open source advantage into GPU vendor libraries thereby, continuing to embark on the same ethos as the deep learning community.

JehandadKhan commented on docs/tutorials/what-is-MIOpen.rst:
```diff
@@ -0,0 +1,17 @@
+
+MIOpen is AMD’s deep learning primitives library, which provides highly optimized, and hand-tuned implementations of
+different operators such as convolution, batch normalization, pooling, softmax, activation and layers for Recurrent Neural
+Networks (RNNs), used in both training and inference. Moreover, MIOpen is fully open-source including all its
+GPU kernels; complementing AMD’s open source ROCm stack [10]. MIOpen is the first to extend the open source
+advantage into GPU vendor libraries thereby, continuing to embark on the same ethos as the deep learning community.
```
```suggestion
Networks (RNNs), used in both training and inference.
```

JehandadKhan reviewed (changes requested):
Minor edits

samjwu commented:
> @samjwu Using 'caption' breaks the build. Without the use of 'caption', it builds fine. Please advise.

Please refer to this version of the toc. It has the correct formatting to build the documentation

https://github.com/ROCm/MIOpen/blob/cac7a8872707c117d88cb56993470c7f371809cf/docs/sphinx/_toc.yml.in

Rmalavally committed:
Update what-is-MIOpen.rst

Rmalavally committed:
Update what-is-MIOpen.rst

Minor tweaks

Rmalavally commented:
> > @samjwu Using 'caption' breaks the build. Without the use of 'caption', it builds fine. Please advise.
> 
> Please refer to this version of the toc. It has the correct formatting to build the documentation
> 
> https://github.com/ROCm/MIOpen/blob/cac7a8872707c117d88cb56993470c7f371809cf/docs/sphinx/_toc.yml.in

If I use hipSPARSElt as the template, then a title is required for each file. The updated ToC with titles builds fine as long as caption is not used. 

https://github.com/ROCm/hipSPARSELt/blob/develop/docs/sphinx/_toc.yml.in

LisaDelaney commented:
> > > @samjwu Using 'caption' breaks the build. Without the use of 'caption', it builds fine. Please advise.
> > 
> > 
> > Please refer to this version of the toc. It has the correct formatting to build the documentation
> > https://github.com/ROCm/MIOpen/blob/cac7a8872707c117d88cb56993470c7f371809cf/docs/sphinx/_toc.yml.in
> 
> If I use hipSPARSElt as the template, then a title is required for each file. The updated ToC with titles builds fine as long as caption is not used.
> 
> https://github.com/ROCm/hipSPARSELt/blob/develop/docs/sphinx/_toc.yml.in

A title is not required...but I like to include it because, if you don't, it uses the title (H1) from within the file, which is often too long for a toc. You can find more info on correct syntax here: https://sphinx-external-toc.readthedocs.io/en/latest/user_guide/sphinx.html#basic-structure. Note that YAML is very particular about spacing/tabs.

LisaDelaney commented on docs/sphinx/_toc.yml.in:
```diff
@@ -1,24 +1,57 @@
+- caption: API reference
+  subtrees:
+    - entries
+      - file: reference/apireference.rst
```
You're referencing a lot of files/folders that don't exit...and need to include the API ref from the doxygen folder (see my previous comment).

Rmalavally replied:
Bibek and I added the API ref from the Doxygen folder. @bghimireamd Please advise.

Rmalavally replied:
Exit? or Exist? We have reused all existing and original files. Sam recommended we create the appropriate folders, and as suggested, we used hipSPARSElt as the template. Should we remove the titles and keep it simple for 6.0?

LisaDelaney replied:
that was a typo: I meant exist.

LisaDelaney commented:
@Rmalavally I just noticed there is no gitignore file in this repo--can you create one (it will live in the root directory)? We don't need the build/doxygen folder/other build artifacts being tracked in the GitHub repo. You can use the ROCm repo gitignore as a reference.

Rmalavally commented:
> @Rmalavally I just noticed there is no gitignore file in this repo--can you create one (it will live in the root directory)? We don't need the build/doxygen folder/other build artifacts being tracked in the GitHub repo. You can use the ROCm repo gitignore as a reference.

Let me check. Could you also update the doc guidelines accordingly? For now, we have, 

>work on splitting up the individual files as needed (if applicable) and/or creating any new files you want (e.g. the "What is X?" page)
create any folders you need and put the appropriate files in the appropriate folders
get the toc file set up the way you want
update the index file(s) to align with the toc structure

Rmalavally commented:
> > @Rmalavally I just noticed there is no gitignore file in this repo--can you create one (it will live in the root directory)? We don't need the build/doxygen folder/other build artifacts being tracked in the GitHub repo. You can use the ROCm repo gitignore as a reference.
> 
> Let me check. Could you also update the doc guidelines accordingly? For now, we have,
> 
> > work on splitting up the individual files as needed (if applicable) and/or creating any new files you want (e.g. the "What is X?" page)
> > create any folders you need and put the appropriate files in the appropriate folders
> > get the toc file set up the way you want
> > update the index file(s) to align with the toc structure

The ROCm .gitignore file has _toc.yml. Assume we will have _toc.yml.in in the .gitignore file. Please confirm.

LisaDelaney commented:
> > > @Rmalavally I just noticed there is no gitignore file in this repo--can you create one (it will live in the root directory)? We don't need the build/doxygen folder/other build artifacts being tracked in the GitHub repo. You can use the ROCm repo gitignore as a reference.
> > 
> > 
> > Let me check. Could you also update the doc guidelines accordingly? For now, we have,
> > > work on splitting up the individual files as needed (if applicable) and/or creating any new files you want (e.g. the "What is X?" page)
> > > create any folders you need and put the appropriate files in the appropriate folders
> > > get the toc file set up the way you want
> > > update the index file(s) to align with the toc structure
> 
> The ROCm .gitignore file has _toc.yml. Assume we will have _toc.yml.in in the .gitignore file. Please confirm.

No, _toc.yml is an autogenerated file (a copy of _toc.yml.in);  _toc.yml.in should not be in the gitignore file.

Rmalavally commented:
> > > > @Rmalavally I just noticed there is no gitignore file in this repo--can you create one (it will live in the root directory)? We don't need the build/doxygen folder/other build artifacts being tracked in the GitHub repo. You can use the ROCm repo gitignore as a reference.
> > > 
> > > 
> > > Let me check. Could you also update the doc guidelines accordingly? For now, we have,
> > > > work on splitting up the individual files as needed (if applicable) and/or creating any new files you want (e.g. the "What is X?" page)
> > > > create any folders you need and put the appropriate files in the appropriate folders
> > > > get the toc file set up the way you want
> > > > update the index file(s) to align with the toc structure
> > 
> > 
> > The ROCm .gitignore file has _toc.yml. Assume we will have _toc.yml.in in the .gitignore file. Please confirm.
> 
> No, _toc.yml is an autogenerated file (a copy of _toc.yml.in); _toc.yml.in should not be in the gitignore file.

Ok. Good to know. Thanks for clarifying.

junliume commented:
@Rmalavally this PR is still in draft state, if we are pursuing it to merging, please update it and resolve all the conflicts above. Thanks!

bghimireamd committed:
Rmalavally-patch-1 : edit index.rst file and added install.rst

bghimireamd committed:
Rmalavally-patch-1 : fix code block in install.rst

bghimireamd committed:
Rmalavally-patch-1 : fix  install.rst

bghimireamd committed:
Rmalavally-patch-1 : fix install.rst

bghimireamd committed:
Rmalavally-patch-1 : render code section in install.rst

bghimireamd committed:
Rmalavally-patch-1 : render all code section in install.rst

bghimireamd committed:
Rmalavally-patch-1 : render all code section in install.rst

bghimireamd committed:
Rmalavally-patch-1: fix referencing multiple docs

bghimireamd committed:
Merge branch 'Rmalavally-patch-1' of github.com:ROCm/MIOpen into Rmalavally-patch-1

LisaDelaney commented on .readthedocs.yaml:
```diff
@@ -6,13 +6,13 @@ version: 2
    - requirements: docs/sphinx/requirements.txt
 
 build:
-   os: ubuntu-22.04
```
@samjwu is this change correct?

samjwu reviewed (changes requested):
`.readthedocs.yaml` should not need to be changed. It is for the ReadtheDocs site build configurations
